### DISCUSSION TITLE
NvTrueHDR - RTX HDR for games wont work when streamed in Moonlight / Sunshine

### DISCUSSION THREAD

**Discussion by h107474** [LizardByte/community#393] [Sunshine]
Has anyone tried the new mod that enables fake HDR upscaling of SDR using Nvidia's new RTX HDR feature? Its called NvTrueHDR over on NexusMods. It works a treat locally and as Digital Foundry has said (go check their article on Eurogamer) it beats Microsoft's Auto HDR.

Any ideas why this wont work when using Moonlight game streaming? Moonlight gamestream can do HDR streaming no problem using its Sunshine backend. Windows 11 Auto HDR or native HDR works and games look great on the client.

When I tried this mod, the streamed version was not HDR and the HUD option was not showing the squares on the streamed end even though its clearly in HDR on the client PC's HDR monitor at the same time with the little squares present too.

EDIT:

So looks like this RTX HDR is becoming a supported feature of the new [Nvidia App](https://www.nvidia.com/en-us/geforce/news/nvidia-app-beta-download/). I wonder if it still wont work over a Moonlight stream when they release it officially. If it does not change much technically, I expect it still wont work on a streamed game. What a shame.

**Comment by Darkhost1999** [comment]
This has been discussed and answered over
https://github.com/LizardByte/Sunshine/issues/2163#issuecomment-1962139648

A suggested workaround is to disable RTX HDR specifically for sunshine.exe and available to leave the option enabled globally if you want.

Side note = HDR must be enabled in Windows settings to configure this option inside the Nvidia app.
It takes a minute for the Nvidia app to recognize and notice the change so it's not just toggle the option and bam configurable. It's toggle, wait for Nvidia app to refresh, then configure.

**Comment by ChrisBln2302** [comment]
I also noticed that RTX-HDR is not properly streamed with sunshine/moonlight, despite clearly active on the host-system. 

It doesn't have anything to do with the issue that sunshine is not working when RTX HDR is enabled globally. Even when RTX-HDR is only enabled for a specific game, the hdr-signal somehow get's lost in the stream.

**Reply by h107474** [reply]
Exactly this! Thank you for explaining it perfectly. We need the ability to use RTX HDR and have the stream on the client display this correctly. Why isn't Sunshine capturing the RTX HDR image properly? It captures Windows Auto HDR fine.

**Reply by Darkhost1999** [reply]
I'm pretty sure it doesn't capture it because it's disabled for sunshine.exe lol.
Wait for a fix on the issue linked.

**Reply by h107474** [reply]
Why does it need to be enabled for Sunshine? The game exe is being converted to HDR as can be seen on the host monitor and Sunshine just needs to capture the image the same way is does for games that are enhanced with Windows 11 Auto HDR. Enabling it for the Sunshine doesn't make sense as Sunshine is not a game or video player that needs the HDR conversion. Its already converting the game exe to HDR. 

I'm no expert so please explain how enabling it for Sunshine will allow Sunshine to transmit the game/screen that is already converted to HDR by RTX HDR?

**Reply by billsoupcanlabs** [reply]
This is still an issue after the restart fix, with RTX HDR enabled it will not stream over Sunshine, but only display on the local display.  I also encounter issues enabling if I use an EDID dongle vs. plugging in my HDR-supported monitor directly, where the Nvidia App generates an error, and I have troubles getting the monitor to display unless I kill the Sunshine process completely.  I'll look to see if I can see anything in the logs, however so far everything seems to operate normally from Sunshine's perspective...I think the issue is on Nvidia's side with the RTX HDR filter.  The Dynamic Vibrance filter works no problem.

**Reply by ChrisBln2302** [reply]
I can confirm that streaming of RTX-HDR is still an issue after the restart-fix. AutoHDR works without problems, but RTX-HDR somehow doesn't work. Hope this can be fixed soon.

**Reply by cgutman** [reply]
We don't really control what Windows does or does not capture. Some things like Auto HDR do get captured by the DXGI Desktop Duplication API, while others happen deeper in the driver or display pipeline (AFMF, RTX HDR) which don't get captured.

We can see if Windows.Graphics.Capture API from #2149 can capture it, but if not, there's nothing really we can do to solve this. It would be a feature request to Nvidia to support display capture of RTX HDR effects (which may be a substantial engineering effort for them).

**Reply by ChrisBln2302** [reply]
@cgutman Thank you for this helpful explanation. Then let's hope that the mentioned Windows-API can capture RTX-HDR. If not at least AutoHDR can be used as a fallback in some games.

**Reply by dylanfrankcom** [reply]
After building and trying out #2149, it doesn't seem to capture RTX-HDR in its current iteration.

**Reply by ChrisBln2302** [reply]
That's a shame. So i guess we have to wait for another solution. But thanks anyway for testing.

**Comment by ReenigneArcher** [comment]


**Reply by h107474** [reply]
Not fixed. This is not the problem being discussed here. See above why.

[a comment by an automated account is not shown]
